### ISSUE TITLE
Hacking ID scan is non functional on the nif vendor.

### ISSUE THREAD

**Issue by Dankman** [VOREStation/VOREStation#1761]
#### Brief description of the issue
breaking access requirements on the NIFsoft requires an E-mag

#### What you expected to happen
Pulse the ID scan wire, and be able to buy things that would normally not be available to my departmental access.

#### What actually happened

```
//Specific soft access checking
			var/datum/nifsoft/path = R.item_path
			if(initial(path.access))
				var/list/soft_access = list(initial(path.access))
				var/list/usr_access = usr.GetAccess()
				if(!has_access(soft_access, list(), usr_access) && !emagged )
					usr << "<span class='warning'>You aren't authorized to buy [initial(path.name)].</span>"
					flick(icon_deny,src)
					return

```
This bit here makes it so that you get access denied when you try to buy something you shouldn't normally have access to, in order to fix the problem, the line has to be changed to this.

`	if(!has_access(soft_access, list(), usr_access) && !emagged && scan_id)`

#### Steps to reproduce
Hack ID scan on vending machine, ID scan isn't disabled.

#### Code Revision
https://github.com/VOREStation/VOREStation/commit/256e1db4d6e43368e3e53cf3027e5ae395abdacd
 
#### Anything else you may wish to add:

The emag check is likely a balance thing, because of things like the little two shot gun you can get that costs 2200 credits.

That said, the check itself makes the vending machine inconsistent with every other vendor in the game.

All emagging a vending machine does is toggle IDscan and toggle Contraband. Which is something that can be done with tools.

Basically, because the code check to determine whether or not you can buy things is tied to the emagged variable instead of the id_scan variable, hacking the vending machine for ID scan does absolutely nothing.

Which is both inconsistent with every other vendor in the game, and inconsistent with how emags work in relation to vending machines.

**Comment by Zeldazackman** [comment]
Still a bug in 2020

**Comment by KasparoVy** [comment]
Resolved
